tasks/managed/embargo-check/embargo-check.yaml

```diff
@@ -62,19 +71,56 @@ spec:
             exit 1
         fi
 
+        set +x # There are a lot of custom fields, don't want to pollute the logs
+        CUSTOM_RH_FIELDS="$(curl -H "Authorization: Bearer $ACCESS_TOKEN" https://issues.redhat.com/rest/api/2/field)"
+        set -x
+        CVE_FIELD="$(jq -r '.[] | select(.name=="CVE ID") | .id' <<< "$CUSTOM_RH_FIELDS")"
+        COMPONENT_FIELD="$(jq -r '.[] | select(.name=="Downstream Component Name") | .id' <<< "$CUSTOM_RH_FIELDS")"
+
         RC=0
 
         NUM_ISSUES=$(jq -cr '.releaseNotes.issues.fixed | length' "${DATA_FILE}")
         for ((i = 0; i < NUM_ISSUES; i++)); do
             issue=$(jq -c --argjson i "$i" '.releaseNotes.issues.fixed[$i]' "${DATA_FILE}")
             server=$(jq -r '.source' <<< "$issue")
             API=$(jq -r '.[] | select(.servers[] | contains("'"$server"'")) | .api' <<< "$SUPPORTED_ISSUE_TRACKERS")
-            API_URL="https://$(jq -r '.source' <<< "$issue")/${API}/$(jq -r '.id' <<< "$issue")"
-            curl --fail --output /dev/null "${API_URL}"
+            API_URL="https://"
+            # Add authentication for issues.redhat.com
+            if [ "$server" = "issues.redhat.com" ] ; then
+                API_URL="-H \"Authorization: Bearer $ACCESS_TOKEN\" ${API_URL}"
```

Comment:
At this point I think `API_URL` is no longer a good name, since it contains args. So maybe `CURL_ARGS`? But also, I wonder how this can work with `curl --fail "${API_URL}"`? Because the `""`, it would be one long argument. Instead, a bash array should be used like we do elsewhere.

Actually, I think you would be better off leaving `API_URL` for just the url. And then do:
```
AUTH_ARGS=()
if [ "$server" = "issues.redhat.com" ] ; then
  AUTH_ARGS=(-H "Authorization: Bearer $ACCESS_TOKEN")
fi
...
OUTPUT=$(curl --fail "${AUTH_ARGS[@]}" "${API_URL}")
```

Reply:
Also, both the `AUTH_ARGS=` and `OUTPUT=` would leak the token, so you'll need `set +x` here.

Reply:
Updated